templates/angular/igx-ts/projects/side-nav-auth/files/src/app/authentication/services/fake-backend.service.ts

```diff
@@ -124,11 +125,11 @@ export class BackendInterceptor implements HttpInterceptor {
 
     /** Note: This is used for example purposes only and does NOT generate a valid JWT w/ base64Url encoding */
     private generateToken(payload: any): string {
-        const header = JSON.stringify({
+        const header = {
             alg: 'Mock',
             typ: 'JWT'
-        });
-        return btoa(JSON.stringify(header)) + '.' + btoa(JSON.stringify(payload)) + '.mockSignature';
+        };
+        return btoa(encodeBase64Url(header)) + '.' + btoa(encodeBase64Url(payload)) + '.mockSignature';
```

Comment:
Either `encodeBase64Url` does the actual `btoa` base 64 encoding.. or it should probably be named something like encode non-ASCII chars or something